autoPyTorch/api/base_task.py

```diff
@@ -1419,27 +1382,90 @@ def refit(
-
+        model_configs = []
         for identifier in self.models_:
             model = self.models_[identifier]
-            # this updates the model inplace, it can then later be used in
-            # predict method
-
-            # try to fit the model. If it fails, shuffle the data. This
-            # could alleviate the problem in algorithms that depend on
-            # the ordering of the data.
-            X = self._get_fit_dictionary(
-                dataset_properties=copy.copy(dataset_properties),
-                dataset=dataset,
-                split_id=split_id)
-            fit_and_suppress_warnings(self._logger, model, X, y=None)
+            budget = identifier[-1]  # identifier is seed, num_run, budget
+            model_configs.append((model.config, budget))
+        
+        self._logger.debug(f"Refitting {model_configs}")
+        run_history, _ = run_models_on_dataset(
+            time_left=total_walltime_limit,
+            func_eval_time_limit_secs=run_time_limit_secs,
+            model_configs=model_configs,
+            logger=self._logger,
+            logger_port=self._logger_port,
+            metric=metric,
+            dask_client=self._dask_client,
+            backend=self._backend,
+            memory_limit=memory_limit,
+            disable_file_output=disable_file_output,
+            all_supported_metrics=all_supported_metrics,
+            include=self.include_components,
+            exclude=self.exclude_components,
+            search_space_updates=self.search_space_updates,
+            pipeline_options=pipeline_options,
+            seed=self.seed,
+            multiprocessing_context=self._multiprocessing_context,
+            n_jobs=self.n_jobs,
+            current_search_space=self.search_space,
+            smac_initial_run=self._backend.get_next_num_run()
+        )
+        replace_old_identifiers_to_refit_identifiers = {}
+
+        self._logger.debug(f"Finished refit training")
+        old_identifier_index = None
+        for run_key, run_value in run_history.data.items():
+            config = run_value.additional_info['configuration']
+            refit_identifier = (self.seed, run_value.additional_info['num_run'], run_key.budget)
+            for i, (configuration, _) in enumerate(model_configs):
+                if isinstance(configuration, Configuration):
+                    configuration = configuration.get_dictionary()
+                self._logger.debug(f"Matching {config} with {configuration}")
+                if config == configuration:
+                    old_identifier_index = i
+                    break
+            if old_identifier_index is not None:
+                replace_old_identifiers_to_refit_identifiers[list(self.models_.keys())[old_identifier_index]] = refit_identifier
+            else:
+                self._logger.warning(f"Refit for {config} failed. Updating ensemble weights accordingly.")
```

Comment:
Do we still update the ensemble weights?

Reply:
thanks, I have fixed it